src/components/interactive-builder/interactive-builder.component.tsx

```diff
@@ -204,15 +209,134 @@ const InteractiveBuilder: React.FC<InteractiveBuilderProps> = ({
     [onSchemaChange, schema, t]
   );
 
+  const handleFormValidation = () => {
+    setResponses([]);
+
+    if (schema) {
+      const parsedForm =
+        typeof schema == "string" ? JSON.parse(schema) : schema;
+
+      for (let i = 0; i < parsedForm.pages.length; i++) {
+        for (let j = 0; j < parsedForm.pages[i].sections.length; j++) {
+          for (
+            let k = 0;
+            k < parsedForm.pages[i].sections[j].questions.length;
+            k++
+          ) {
+            const questionObject = parsedForm.pages[i].sections[j].questions[k];
+
+            if (questionObject.type === "obsGroup") {
+              questionObject.questions.forEach((question) => {
+                handleQuestionValidation(question);
+              });
+            }
+            handleQuestionValidation(questionObject);
+          }
+        }
+      }
+    } else {
+      console.log("Empty form!");
+    }
+  };
+
+  const handleQuestionValidation = (conceptObject) => {
+    const conceptRepresentation =
+      "custom:(uuid,display,datatype,conceptMappings:(conceptReferenceTerm:(conceptSource:(name),code)))";
+
+    const searchRef = conceptObject.questionOptions.concept
+      ? conceptObject.questionOptions.concept
+      : conceptObject.questionOptions.conceptMappings
+      ? conceptObject.questionOptions.conceptMappings
+          ?.map((mapping) => {
+            return `${mapping.type}:${mapping.value}`;
+          })
+          .join(",")
+      : "";
+
+    searchRef
+      ? openmrsFetch(
+          `/ws/rest/v1/concept?references=${searchRef}&v=${conceptRepresentation}`
+        )
+          .then((response) => {
+            if (response.data.results.length > 0) {
+              const [resObject] = response.data.results;
+
+              dataTypeChecker(conceptObject, resObject);
+            } else {
+              resolver(
+                conceptObject,
+                `Concept "${conceptObject.questionOptions.concept}" not found`
+              );
+            }
+          })
+          .catch((error) => {
+            console.log(error.message);
+          })
```

Comment:
Wrap this logic with an API function and harbor it within the API file. This ensures separation of concerns.

Reply:
Noted.